ArcGISRuntimeSDKQt_CppSamples/UtilityNetwork/CreateLoadReport/CreateLoadReport.cpp

```diff
@@ -252,34 +252,29 @@ void CreateLoadReport::setUtilityTraceOrconditionWithCodedValue(CodedValue coded
   m_traceParameters->traceConfiguration()->traversability()->setBarriers(utilityTraceOrCondition);
 }
 
-void CreateLoadReport::createTraceCompletedConnection()
+void CreateLoadReport::createTraceCompletedConnection(const QString& codedValueName)
 {
-  connect(m_utilityNetwork, &UtilityNetwork::traceCompleted, this, [this](const QUuid& taskId)
-  {
-    const QString codedValueName = m_tasks.take(taskId);
-
-    UtilityTraceResultListModel* results = m_utilityNetwork->traceResult();
+  UtilityTraceResultListModel* results = m_utilityNetwork->traceResult();
 
-    for (UtilityTraceResult* result : *results)
-    {
-      // Get the total customers from the UtilityElementTraceResult
-      if (UtilityElementTraceResult* elementResult = dynamic_cast<UtilityElementTraceResult*>(result))
-        m_phaseCust[codedValueName] = elementResult->elements(this).size();
+  for (UtilityTraceResult* result : *results)
+  {
+    // Get the total customers from the UtilityElementTraceResult
+    if (UtilityElementTraceResult* elementResult = dynamic_cast<UtilityElementTraceResult*>(result))
+      m_phaseCust[codedValueName] = elementResult->elements(this).size();
 
-      // Get the total load from the UtilityFunctionTraceResult
-      else if (UtilityFunctionTraceResult* functionResult = dynamic_cast<UtilityFunctionTraceResult*>(result))
-        m_phaseLoad[codedValueName] = qAsConst(functionResult)->functionOutputs().first()->result().toInt();
-    }
+    // Get the total load from the UtilityFunctionTraceResult
+    else if (UtilityFunctionTraceResult* functionResult = dynamic_cast<UtilityFunctionTraceResult*>(result))
+      m_phaseLoad[codedValueName] = qAsConst(functionResult)->functionOutputs().first()->result().toInt();
+  }
 
-    emit loadReportUpdated();
+  emit loadReportUpdated();
 
-    // If the tasks queue is empty, all trace tasks have completed
-    if (m_tasks.isEmpty())
-    {
-      m_sampleStatus = CreateLoadReport::SampleReady;
-      emit sampleStatusChanged();
-    }
-  });
+  // If the tasks queue is empty, all trace tasks have completed
+  if (m_tasks.isEmpty())
```

Comment:
I don't think this makes sense anymore, either. `m_tasks` is a map of QUuids, which are probably the old TaskWatcher ids. It should be updated to the new pattern.

Reply:
Sure. I am working on this